src/test/java/com/ddudu/goal/controller/GoalControllerTest.java

```diff
@@ -28,169 +33,226 @@
 import org.springframework.boot.test.autoconfigure.web.servlet.WebMvcTest;
 import org.springframework.boot.test.mock.mockito.MockBean;
 import org.springframework.context.annotation.Import;
 import org.springframework.http.MediaType;
+import org.springframework.security.oauth2.jose.jws.MacAlgorithm;
+import org.springframework.security.oauth2.jwt.JwsHeader;
+import org.springframework.security.oauth2.jwt.Jwt;
+import org.springframework.security.oauth2.jwt.JwtClaimsSet;
+import org.springframework.security.oauth2.jwt.JwtEncoder;
+import org.springframework.security.oauth2.jwt.JwtEncoderParameters;
 import org.springframework.test.web.servlet.MockMvc;
+import org.springframework.test.web.servlet.ResultActions;
 
 @WebMvcTest(controllers = GoalController.class)
 @Import({WebSecurityConfig.class, TestProperties.class, JwtConfig.class})
 @DisplayNameGeneration(ReplaceUnderscores.class)
 class GoalControllerTest {
 
   static final Faker faker = new Faker();
+  static final JwsHeader header = JwsHeader.with(MacAlgorithm.HS512)
+      .build();
+  static final JwtClaimsSet.Builder claimSet = JwtClaimsSet.builder()
+      .claim("auth", Authority.NORMAL);
 
   @Autowired
   private MockMvc mockMvc;
-
   @Autowired
   private ObjectMapper objectMapper;
+  @Autowired
+  private JwtEncoder jwtEncoder;
 
   @MockBean
   private GoalService goalService;
 
-  String validName;
-  String validColor;
+  private String validName;
+  private Long validUserId;
+
+  private String validColor;
+  private PrivacyType validPrivacy;
 
   @BeforeEach
   void setUp() {
     validName = faker.lorem()
         .word();
+    validUserId = faker.random()
+        .nextLong();
     validColor = faker.color()
         .hex()
         .substring(1);
+    validPrivacy = provideRandomPrivacy();
   }
 
   @Nested
   class POST_목표_생성_API_테스트 {
 
     @Test
-    void 목표_생성을_성공한다() throws Exception {
+    void 목표_생성에_성공하면_201_Created_응답을_반환한다() throws Exception {
       // given
-      CreateGoalRequest request = new CreateGoalRequest(validName, validColor, PrivacyType.PUBLIC);
-      CreateGoalResponse response = new CreateGoalResponse(1L, validName, validColor);
+      Long goalId = faker.random()
+          .nextLong();
+      CreateGoalRequest request = new CreateGoalRequest(validName, validColor, validPrivacy);
+      CreateGoalResponse response = new CreateGoalResponse(goalId, validName, validColor);
 
       given(goalService.create(anyLong(), any(CreateGoalRequest.class)))
           .willReturn(response);
 
-      // when then
-      mockMvc.perform(
-              post("/api/goals")
-                  .param("userId", "1")
-                  .content(objectMapper.writeValueAsString(request))
-                  .contentType(MediaType.APPLICATION_JSON)
-          )
-          .andExpect(status().isCreated())
+      // when
+      ResultActions actions = mockMvc.perform(
+          post("/api/goals")
+              .header("Authorization", getToken(validUserId))
```

Comment:
`String token` 변수를 상단에 두고 **@beforeEach**에서 `token = createBearerToken or getToken(validUserId);` 이렇게 설정하고, 매개변수를 token으로 설정하는 건 어떻게 생각하시나용?

```suggestion
              .header("Authorization", token)
```

Reply:
좋은 것 같습니당! 바로 적용해 놓을게용!!